carp.common/src/commonMain/kotlin/dk/cachet/carp/common/infrastructure/versioning/Major1Minor1To3Migration.kt

```diff
@@ -0,0 +1,23 @@
+package dk.cachet.carp.common.infrastructure.versioning
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.JsonPrimitive
+
+
+/**
+ * - The `deviceDisplayName` field in `DeviceRegistration` is no longer required.
+ *   This also apply to `ConnectedDeviceRegistration`.
+ */
+@Suppress( "MagicNumber" )
+abstract class Major1Minor1To3Migration : ApiMigration( 1, 3 )
```

Comment:
Why is this abstract? I'm guessing you got a bit mislead by the `Major1Minor0To1Migration` base class ([also living in this namespace](https://github.com/cph-cachet/carp.core-kotlin/blob/develop/carp.common/src/commonMain/kotlin/dk/cachet/carp/common/infrastructure/versioning/Major1Minor0To1Migration.kt)).

The reason this was abstract, and lived in common, as opposed to a specific subsystem being upgrades, was that it was a cross-cutting concern/upgrade for _all_ subsystems. That is not the case here. So, this should be moved to the respective subsystem this upgrade is related to, and a base class is irrelevant as there is no reuse.

Reply:
I created this abstract class because I believe it's better to have a reference for all the migrations that exist in the entire system. That's also the reason I named the upgrader "version 2 to 3" initially as I was following the release version.

Now I realise that each subsystem is rolling their own ApiVersions, and this naming schema will eventually not work.

Reply:
Yup. 🙂 The reason they are separate subsystems is so that we can evolve them independently. 😉

Reply:
hmmm..the upgrading should then be 1 to 2, by following the `DeploymentService.ApiVersion`
I think my intuition was naming should follow the release version because, the migration will not be published until the next release. It would made more sense if each subsystem are also released separately. (now I think about it, they do release separately, each api migration bumbs api version)

I should really start treating subsystems as separate systems, that caused all the confusion here.

Reply:
> Any strategy enforcing subsystem apiversion to match release version will make each subsystems' own apiversion meaningless.

Not really. Each time there is _a_ release of CARP core, a new release for all subsystems can be done, regardless of whether there are any changes in that subsystem. For the time being, that simplifies release management/deployment. At most, it's a minor overhead.

But, we wouldn't want to unnecessarily bump API versions. So at any given release, the API version may run behind if there were no new features added to an endpoint.

That would then mean that from time to time (like potentially now), if new features are added, you'd simply jump a couple of minor versions instead of 1.

Reply:
Thanks for the explanation! I also found the discussion in #467 helpful, which I didn't notice in the first place.